### ISSUE TITLE
Folium 0.4.0 gives serialization error using an ImageOverlay with file source

### ISSUE THREAD

**Issue by dirkvdb** [python-visualization/folium#709]
This code is broken in release 0.4.0

```
import folium
import folium.plugins

map = folium.Map(crs="EPSG3857", tiles="OpenStreetMap");
overlay = folium.plugins.ImageOverlay(open("c:\\Temp\raster.png", "br"), opacity=0.7, bounds=[[2.210320729283855, -85.29479965127808], [1.3832410843731593, -83.16591433255314]]);
```
It gives the following exception
```
c:\python36\lib\site-packages\folium\plugins\image_overlay.py in __init__(self, image, bounds, origin, colormap, mercator_project, overlay, control, pixelated, name, **kwargs)
     78                 origin=origin)
     79 
---> 80         self.url = image_to_url(image, origin=origin, colormap=colormap)
     81 
     82         self.bounds = json.loads(json.dumps(bounds))

c:\python36\lib\site-packages\folium\utilities.py in image_to_url(image, colormap, origin)
    140     else:
    141         # Round-trip to ensure a nice formatted json.
--> 142         url = json.loads(json.dumps(image))
    143     return url.replace('\n', ' ')
    144 

c:\python36\lib\json\__init__.py in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)
    229         cls is None and indent is None and separators is None and
    230         default is None and not sort_keys and not kw):
--> 231         return _default_encoder.encode(obj)
    232     if cls is None:
    233         cls = JSONEncoder

c:\python36\lib\json\encoder.py in encode(self, o)
    197         # exceptions aren't as detailed.  The list call should be roughly
    198         # equivalent to the PySequence_Fast that ''.join() would do.
--> 199         chunks = self.iterencode(o, _one_shot=True)
    200         if not isinstance(chunks, (list, tuple)):
    201             chunks = list(chunks)

c:\python36\lib\json\encoder.py in iterencode(self, o, _one_shot)
    255                 self.key_separator, self.item_separator, self.sort_keys,
    256                 self.skipkeys, _one_shot)
--> 257         return _iterencode(o, 0)
    258 
    259 def _make_iterencode(markers, _default, _encoder, _indent, _floatstr,

c:\python36\lib\json\encoder.py in default(self, o)
    178         """
    179         raise TypeError("Object of type '%s' is not JSON serializable" %
--> 180                         o.__class__.__name__)
    181 
    182     def encode(self, o):

TypeError: Object of type 'BufferedReader' is not JSON serializable
```

0.3.0 works fine. Is a file source no longer supported?

**Comment by ocefpaf** [comment]
> 0.3.0 works fine. Is a file source no longer supported?

Actually a file source should be the only supported way now. (I forgot to add that to the API changes.)
No need to pass the buffered image, `ImageOverlay('c:\\Temp\raster.png')` should work.

The reason is that in that past when a buffer was passed we embedded the while when file name was passed we did not. I don't think that is a good `API`. For now I am working on making everything embedded by default and later I plan to add a saving option `standalone=[True/False]` (embedded or not).
